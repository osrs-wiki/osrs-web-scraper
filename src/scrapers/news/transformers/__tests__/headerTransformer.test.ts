import {
  MediaWikiBreak,
  MediaWikiBuilder,
  MediaWikiContent,
  MediaWikiText,
  MediaWikiTOC,
} from "@osrs-wiki/mediawiki-builder";

import NewsHeaderTransformer from "../headerTransformer";

describe("NewsHeaderTransformer", () => {
  it("should change bold text to header-3", () => {
    const originalContent: MediaWikiContent[] = [
      new MediaWikiTOC(),
      new MediaWikiBreak(),
      new MediaWikiText(" You can also discuss this update on our ", {
        bold: true,
      }),
      new MediaWikiBreak(),
      new MediaWikiText("The Old School Team.", { italics: true }),
    ];
    const transformed = new NewsHeaderTransformer().transform(originalContent);
    expect(
      new MediaWikiBuilder().addContents(transformed).build()
    ).toMatchSnapshot();
  });

  it("should not change bold text to header-3", () => {
    const originalContent: MediaWikiContent[] = [
      new MediaWikiTOC(),
      new MediaWikiText("test", { bold: true }),
      new MediaWikiBreak(),
      new MediaWikiText("The Old School Team.", { italics: true }),
    ];
    const transformed = new NewsHeaderTransformer().transform(originalContent);
    expect(
      new MediaWikiBuilder().addContents(transformed).build()
    ).toMatchSnapshot();
  });
});
