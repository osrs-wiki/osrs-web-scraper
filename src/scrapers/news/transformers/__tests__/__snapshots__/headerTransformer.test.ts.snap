// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsHeaderTransformer should change bold text to header-3 1`] = `
"__TOC__

====You can also discuss this update on our====
''The Old School Team.''"
`;

exports[`NewsHeaderTransformer should not change bold text to header-3 1`] = `
"__TOC__
'''test'''
''The Old School Team.''"
`;
